synapse/storage/databases/main/event_push_actions.py

```diff
@@ -1340,6 +1349,53 @@ def remove_old_push_actions_that_have_rotated_txn(
             if done:
                 break
 
+    @wrap_as_background_process("_clear_old_push_actions_staging")
+    async def _clear_old_push_actions_staging(self) -> None:
+        """Clear out any old event push actions from the staging table for
+        events that we failed to persist.
+        """
+
+        # We delete anything more than an hour old, on the assumption that we'll
+        # never take more than an hour to persist an event.
+        delete_before_ts = self._clock.time_msec() - 60 * 60 * 1000
```

Comment:
I think this is building in an assumption that the background worker and event persister workers are restarted within 1 hour of each other.

Reply:
Ah, no, since we add a default to `inserted_ts` for all new rows, so even if we don't restart the event persister we should have the right values in the table.

(I think, i spent quite a while trying to figure out how to make this work)